src/fastalloc/mod.rs

```diff
@@ -1209,11 +1209,35 @@ impl<'a, F: Function> Env<'a, F> {
         Ok(())
     }
 
+    fn point_range_intersect(
+        &self,
+        (start_inst, end_inst): (Inst, Inst),
+        (point_start, point_end): (ProgPoint, ProgPoint),
+    ) -> Option<(ProgPoint, ProgPoint)> {
+        if end_inst <= point_start.inst() || start_inst >= point_end.inst() {
+            None
+        } else {
+            let point0 = if point_start.inst() >= start_inst {
```

Comment:
`ProgPoint` implements `Ord`, so could we use `std::cmp::min` / `std::cmp::max` here?